frontend/appflowy_flutter/lib/plugins/database_view/grid/presentation/widgets/header/field_editor.dart

```diff
@@ -240,7 +209,9 @@ class DeleteFieldButton extends StatelessWidget {
     return BlocBuilder<FieldEditorBloc, FieldEditorState>(
       buildWhen: (previous, current) => previous != current,
       builder: (context, state) {
-        final enable = !state.canDelete && !state.isGroupField;
+        final enable = state.field != null &&
```

Comment:
The state can expose a property that determines whether the button is enabled or not. We can then encapsulate the logic within the bloc

Reply:
Sounds good